external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -3271,697 +3233,51 @@ fn parse_module_member(context: &mut Context) -> Result<ModuleMember, ErrCase> {
-            name,
-            post_state,
-            def,
-        },
-    ))
-}
-
-// Parse a specification schema include.
-//    SpecInclude = "include" <Exp>
-fn parse_spec_include(context: &mut Context) -> Result<SpecBlockMember, Box<Diagnostic>> {
-    let start_loc = context.tokens.start_loc();
-    consume_identifier(context.tokens, "include")?;
-    let properties = parse_condition_properties(context)?;
-    let exp = parse_exp(context)?;
-    consume_token(context.tokens, Tok::Semicolon)?;
-    Ok(spanned(
-        context.tokens.file_hash(),
-        start_loc,
-        context.tokens.previous_end_loc(),
-        SpecBlockMember_::Include { properties, exp },
-    ))
-}
-
-// Parse a specification schema apply.
-//    SpecApply = "apply" <Exp> "to" Comma<SpecApplyPattern>
-//                                   ( "except" Comma<SpecApplyPattern> )? ";"
-fn parse_spec_apply(context: &mut Context) -> Result<SpecBlockMember, Box<Diagnostic>> {
-    let start_loc = context.tokens.start_loc();
-    consume_identifier(context.tokens, "apply")?;
-    let exp = parse_exp(context)?;
-    consume_identifier(context.tokens, "to")?;
-    let parse_patterns = |context: &mut Context| {
-        parse_list(
-            context,
-            |context| {
-                if context.tokens.peek() == Tok::Comma {
-                    context.tokens.advance()?;
-                    Ok(true)
-                } else {
-                    Ok(false)
-                }
-            },
-            parse_spec_apply_pattern,
-        )
-    };
-    let patterns = parse_patterns(context)?;
-    let exclusion_patterns =
-        if context.tokens.peek() == Tok::Identifier && context.tokens.content() == "except" {
-            context.tokens.advance()?;
-            parse_patterns(context)?
-        } else {
-            vec![]
-        };
-    consume_token(context.tokens, Tok::Semicolon)?;
-    Ok(spanned(
-        context.tokens.file_hash(),
-        start_loc,
-        context.tokens.previous_end_loc(),
-        SpecBlockMember_::Apply {
-            exp,
-            patterns,
-            exclusion_patterns,
-        },
-    ))
-}
+    }
 
-// Parse a function pattern:
-//     SpecApplyPattern = <SpecApplyFragment>+ <OptionalTypeArgs>
-fn parse_spec_apply_pattern(context: &mut Context) -> Result<SpecApplyPattern, Box<Diagnostic>> {
-    let start_loc = context.tokens.start_loc();
-    // TODO: update the visibility parsing in the spec as well
-    let public_opt = consume_optional_token_with_loc(context.tokens, Tok::Public)?;
-    let visibility = if let Some(loc) = public_opt {
-        Some(Visibility::Public(loc))
-    } else if context.tokens.peek() == Tok::Identifier && context.tokens.content() == "internal" {
-        // Its not ideal right now that we do not have a loc here, but acceptable for what
-        // we are doing with this in specs.
+    // consume semi-colon if present
+    if context.tokens.peek() == Tok::Semicolon {
```

Comment:
Should we put this behind a flag or something? I think it will allow a semicolon in some additional spots that it wasn't before. Not that its a big deal either way

Reply:
I think it's probably fine? Only occurs if you're using specs (and writing more after this too, which is kinda???).